common/lib/common-utils/src/hashFileBrowser.ts

```diff
@@ -5,20 +5,31 @@
 
 import { IsoBuffer } from "./bufferBrowser";
 
+let insecureContextHashFn: ((f: IsoBuffer) => Promise<string>) | undefined;
+
+/**
+ * Set a hashing function to be called in place of hashFile's internal
+ * implementation when running under insecure contexts.  Not needed
+ * when running under Node.
+ * @param hashFn - The function that should be used in place of hashFile
+ */
+export function setInsecureContextHashFn(hashFn: (f: IsoBuffer) => Promise<string>) {
```

Comment:
Does the framework require this to be a specific hash function, like SHA? Or does it not care about the hash algorithm

Reply:
it needs to be sha1 right now since that's what's used internally (and because `gitHashFile` uses sha1 and it calls `hashFile` which doesn't let you specify the algorithm).  it _could_ be something else, but that adds extra handling for a very narrow developer scenario that isn't needed elsewhere so i don't think it's necessary for now

Reply:
I was asking more so we can make sure the person providing this knows it has to be sha1